src/util/reverse_range.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
```

Comment:
Nice! However please follow the conventions used in other headers of this project:

1. No subfolder
2. Macro-based header guard instead of pragma once
3. Newline before the opening brace of multiline functions
4. Newline after `template<typename R>`
5. Run `./prepare.rb` to add the license preamble

Reply:
Sure, I'll change the code accordingly, regarding the formatting: maybe you could provide a .clang-format with a preferred style so this part could be handled automatically?

As for the license, this thing can be done solely with CMake. Maybe I should remove Ruby? Using another programming language for concatenating files looks like an overkill.